tools/linker/MarkNSObjects.cs

```diff
@@ -151,9 +152,15 @@ static bool IsProductMethod (MethodDefinition method)
 			return (method.DeclaringType.Module.Assembly.Name.Name == ProductAssembly);
 		}
 
-		static bool IsProductType (TypeDefinition type)
+		bool IsProductType (TypeDefinition type)
 		{
-			return type.Module.Assembly.Name.Name == ProductAssembly;
+			var name = type.Module.Assembly.Name.Name;
+			switch (name) {
+			case "Xamarin.Forms.Platform.iOS":
+				return LinkContext.App.Optimizations.ExperimentalFormsProductType == true;
+			default:
+				return name == ProductAssembly;
+			}
```

Comment:
Oh! You misses the chance to use the new C# switch expressions!

Reply:
I have not yet looked at the IL yet - so I'm sticking to known patterns ;-)